File: src/primus/pawel/Exercise_107.java
package primus.pawel;

import java.util.HashMap;
import java.util.Map;

// Sum of array singles
// 7kyu
// https://www.codewars.com/kata/59f11118a5e129e591000134/train/java
// 2021-05-05 T:20:32:04
class Exercise_107 {

    public static int repeats(int[] arr) {
        int sum = 0;
        HashMap<Integer, Integer> map = new HashMap<>();
        for (int value : arr) {
            if (!map.containsKey(value)) {
                map.put(value, 1);
            } else {
                map.replace(value, map.get(value) + 1);
            }
        }

        for (Map.Entry<Integer, Integer> entry : map.entrySet()) {
            //System.out.println(entry.getKey() + "/" + entry.getValue());
            if (entry.getValue() == 1) {
                sum += entry.getKey();
            }
        }
        return sum;
    }

    public static void main(String[] args) {
        repeats(new int[]{16, 0, 11, 4, 8, 16, 0, 11});
    }
}

File: src/JS/Exercise_107.js

// Sort Numbers
// 7kyu
// https://www.codewars.com/kata/5174a4c0f2769dd8b1000003/train/javascript
// 2022-06-09 T:08:03:01

function solution(nums){
    if(nums == null){
        return []
    }
    return nums.sort((a,b) => a -b)
}

console.log(solution([1,5,6,3]))